tabpfn_client/client.py

```diff
@@ -258,14 +271,52 @@ def predict(
 
         # Send prediction request. Loop two times, such that if anything cached is not correct
         # anymore, there is a second iteration where the datasets are uploaded.
+        results = None
         max_attempts = 2
         for attempt in range(max_attempts):
             try:
-                response = self._make_prediction_request(
+                with self._make_prediction_request(
                     cached_test_set_uid, x_test_serialized, params
-                )
-                self._validate_response(response, "predict")
-                break  # Successful response, exit the retry loop
+                ) as response:
+                    self._validate_response(response, "predict")
+                    # Handle updates from server
+                    client = sseclient.SSEClient(response.iter_bytes())
+
+                    progress_bar = None
+
+                    def run_progress():
+                        nonlocal progress_bar
+                        progress_bar = tqdm(
+                            range(int(duration * 10)),
+                            desc="Processing",
+                            total=int(duration * 10),
+                            bar_format="{desc}: {percentage:3.0f}%|{bar}| [{elapsed}<{remaining}]",
+                        )
+                        for _ in progress_bar:
+                            time.sleep(0.1)
+
+                    for event in client.events():
+                        data = json.loads(event.data)
+                        if data["event"] == "duration_estimate":
+                            duration = float(data["data"])
+                            print(f"Duration estimate: {duration} seconds")
+                            progress_thread = threading.Thread(target=run_progress)
+                            progress_thread.daemon = True
+                            progress_thread.start()
+                        elif data["event"] == "result":
+                            results = data["data"]
+                            if progress_bar:
+                                progress_bar.n = progress_bar.total
+                                progress_bar.refresh()
+                                progress_bar.close()
+                        elif data["event"] == "error":
+                            if data["error_class"] == "GCPOverloaded":
+                                raise GCPOverloaded(data["detail"])
+                            elif data["error_class"] == "ValueError":
+                                raise ValueError(data["detail"])
+                            else:
+                                raise RuntimeError(data["detail"])
```

Comment:
Add "error_class"

Reply:
What do you mean, add it in the RuntimeError message?

Reply:
raise RuntimeError(data["error_class"] + data["detail"])